internal/olm/operator/registry/operator_installer_test.go

```diff
@@ -16,22 +16,233 @@ package registry
+
+			// Test
+			sub := &v1alpha1.Subscription{
+				Status: v1alpha1.SubscriptionStatus{
+					InstallPlanRef: &corev1.ObjectReference{
+						Name:      "fakeName",
+						Namespace: "fakeNS",
+					},
+				},
+			}
+			err = oi.approveInstallPlan(context.TODO(), sub)
+			Expect(err).ToNot(HaveOccurred())
+			err = oi.cfg.Client.Get(context.TODO(), ipKey, ip)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ip.Name).To(Equal("fakeName"))
+			Expect(ip.Namespace).To(Equal("fakeNS"))
+			Expect(ip.Spec.Approved).To(Equal(true))
+		})
+		It("should return an error if the install plan does not exist.", func() {
+			oi.cfg.Client = fake.NewFakeClientWithScheme(sch)
+			sub := &v1alpha1.Subscription{
+				Status: v1alpha1.SubscriptionStatus{
+					InstallPlanRef: &corev1.ObjectReference{
+						Name:      "fakeName",
+						Namespace: "fakeNS",
+					},
+				},
+			}
+			err := oi.approveInstallPlan(context.TODO(), sub)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("error getting install plan"))
+		})
+		// TODO(asmacdo) How to throw an err on update?
+		// It("should return an error if the install plan fails to update.", func() {
+		// 	Expect(err.Error()).Should(ContainSubstring("error approving install plan"))
+		// })
+	})
+
+	Describe("waitForInstallPlan", func() {
+		var (
+			oi  *OperatorInstaller
+			sch *runtime.Scheme
+		)
+		BeforeEach(func() {
+			// Setup and fake client
+			kubeconfigPath := os.Getenv(k8sutil.KubeConfigEnvVar)
+			cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath}
+			sch = runtime.NewScheme()
+			Expect(v1alpha1.AddToScheme(sch)).To(Succeed())
+			cfg.Client = fake.NewFakeClientWithScheme(sch)
+
+			oi = NewOperatorInstaller(cfg)
+			oi.CatalogCreator = FakeCatalogCreator{
+				client: cfg.Client,
+			}
+			oi.StartingCSV = "fakeName"
+			oi.cfg.Namespace = "fakeNS"
+		})
+		It("should return an error if the subscription does not exist.", func() {
+			sub := newSubscription(oi.StartingCSV, oi.cfg.Namespace,
+				withPackageChannel(oi.PackageName, oi.Channel, oi.StartingCSV),
+				withCatalogSource("duplicate", oi.cfg.Namespace),
+				withInstallPlanApproval(v1alpha1.ApprovalManual))
+
+			err := oi.waitForInstallPlan(context.TODO(), sub)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("install plan is not available for the subscription"))
+
+		})
+		It("should return if subscription has an install plan.", func() {
+			sub := &v1alpha1.Subscription{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fakeName",
+					Namespace: "fakeNS",
+				},
+				Status: v1alpha1.SubscriptionStatus{
+					InstallPlanRef: &corev1.ObjectReference{
+						Name:      "fakeName",
+						Namespace: "fakeNS",
+					},
+				},
+			}
+			err := oi.cfg.Client.Create(context.TODO(), sub)
+			Expect(err).ToNot(HaveOccurred())
+			err = oi.waitForInstallPlan(context.TODO(), sub)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
I would probably put a blank line before the `waitForInstallPlan` so that it is easier to see the stuff above it are for setup.